libsel4vm/src/arch/x86/processor/decode.c

```diff
@@ -27,9 +27,17 @@ Author: W.A.
 /* TODO are these defined elsewhere? */
 #define IA32_PDE_SIZE(pde) (pde & BIT(7))
 #define IA32_PDE_PRESENT(pde) (pde & BIT(0))
+
+#ifdef CONFIG_X86_64_VTX_64BIT_GUESTS
```

Comment:
Wouldn't it make more sense to always define it like this if the host is 64-bit? The only reason no to have it like that is for 32-it x86, right?

Reply:
I don't think so. Without `CONFIG_X86_64_VTX_64BIT_GUESTS`, with a 64-bit host you have a 32-bit guest, which cannot have 64-bit memory, so all of the page tables are in low memory where a 32-bit mask is appropriate.

Reply:
But a 32-bit guest can have PAE support too, right?

So not really related to your changes, but who controls what addresses `vm_fetch_instruction()` deals with? That is, are all those assumptions and asserts safe? What if `eip` points to invalid memory?

Reply:
No. 32-bit guests do not have PAE.

![image](https://github.com/seL4/seL4_projects_libs/assets/21088707/eea5d377-2bb4-4f23-ab6d-8c1740aab023)

There's also a check to enforce this:

```
    /* ensure that PAE is not enabled */
#ifndef CONFIG_X86_64_VTX_64BIT_GUESTS
    if (cr4 & X86_CR4_PAE) {
        ZF_LOGE("Do not support walking PAE paging structures");
        return -1;
    }
#endif /* not CONFIG_X86_64_VTX_64BIT_GUESTS */
```

Additionally, the macros deal specifically with physical addresses, which in a 32-bit guest will always be 32-bit.